app/assets/javascripts/application.js

```diff
@@ -338,6 +338,26 @@ $(function() {
   });
 });
 
+function check_for_unavailable_puppetclasses() {
+  var unavailable_classes = $('#puppet_klasses #selected_classes .unavailable');
```

Comment:
I have to agree with @ohadlevy here, the function is relatively complicated so if we can get unit tests for them that's great. It doesn't have to be a React component, just moving the function there would make sense IMO so you can use Jest to test it. Like this one: https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/foreman_users.js